sandbox/test_linux.go

```diff
@@ -0,0 +1,40 @@
+package sandbox
+
+import (
+	"runtime"
+	"syscall"
+	"testing"
+)
+
+// SetupTestNetNS joins a new network namespace, and returns its associated
+// teardown function.
+//
+// Example usage:
+//
+//     defer SetupTestNetNS(t)()
+//
+func setupTestNetNS(t *testing.T) func() {
```

Comment:
Minor comment: We do not really need to keep this internal function. Implementation can all be moved in `SetupTestOSContext()`